op-alt-da/damock.go

```diff
@@ -99,3 +103,88 @@ func (d *AltDADisabled) OnFinalizedHeadSignal(f HeadSignalFn) {
 func (d *AltDADisabled) AdvanceL1Origin(ctx context.Context, l1 L1Fetcher, blockId eth.BlockID) error {
 	return ErrNotEnabled
 }
+
+// FakeDAServer is a fake DA server for e2e tests.
+// It is a small wrapper around DAServer that allows for setting request latencies,
+// to mimic a DA service with slow responses (eg. eigenDA with 10 min batching interval).
+type FakeDAServer struct {
+	*DAServer
+	putRequestLatency time.Duration
+	getRequestLatency time.Duration
+}
+
+func NewFakeDAServer(host string, port int, log log.Logger) *FakeDAServer {
+	store := NewMemStore()
+	fakeDAServer := &FakeDAServer{
+		DAServer:          NewDAServer(host, port, store, log, true),
+		putRequestLatency: 0,
+		getRequestLatency: 0,
+	}
+	return fakeDAServer
+}
+
+func (s *FakeDAServer) HandleGet(w http.ResponseWriter, r *http.Request) {
+	if s.getRequestLatency > 0 {
+		time.Sleep(s.getRequestLatency)
+	}
```

Comment:
We could do without the `if`, since numbers <=0 will make sleep return immediately.

Reply:
Unfortunately not. I wrote the explanation in the comment right above. Only < 0 means no limit. 0 literally means no goroutine can start. See https://github.com/golang/go/issues/27837#issuecomment-1121177106

Reply:
Are you mixing this up with `MaxConcurrentDARequests`? This is just a sleep, no goroutines involved.

Reply:
I was! shouldn't review PRs on my phone haha

Reply:
Fixed: [dbcec39](https://github.com/Layr-Labs/optimism/pull/7/commits/dbcec398525377cb9f212c6ed8b8e3f5abde4695)